### ISSUE TITLE
Standalone app crashes after being in the background for a while

### ISSUE THREAD

**Issue by rotembcohen** [expo/expo#656]
**Issue (TL;DR):**
When opening the standalone Android app, if the app was in the background for sometime, it crashes.

**Device info:**
Device: Samsung s8+ (unrooted)
Android Version: 7.0

**App info:**
App originally created using CRNA
Standalone created using 'exp build:android'
Uploaded to the play store

**Reproduce:**
1. Open the app.
2. Leave the app using home button (at any screen)
3. Open the app after a while, usually an hour or so, you see the expo splash screen, after a few seconds the screen moves a little to the left and the app crashes. Opening the app again works well.

**Error as logged in the Google dev console:**
```
abi20_0_0.com.facebook.react.uimanager.IllegalViewOperationException: 
  at abi20_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.resolveViewManager (NativeViewHierarchyManager.java:131)
  at abi20_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren (NativeViewHierarchyManager.java:294)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded (UIViewOperationQueue.java:813)
  at abi20_0_0.com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:21)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6776)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1520)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1410)`
```
**package.json:**
```
{
  "name": "pocat",
  "version": "1.3.0",
  "private": true,
  "devDependencies": {
    "react-native-scripts": "1.3.1",
    "jest-expo": "~20.0.0",
    "react-test-renderer": "16.0.0-alpha.12"
  },
  "main": "./node_modules/react-native-scripts/build/bin/crna-entry.js",
  "scripts": {
    "start": "react-native-scripts start",
    "eject": "react-native-scripts eject",
    "android": "react-native-scripts android",
    "ios": "react-native-scripts ios",
    "test": "node node_modules/jest/bin/jest.js --watch"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "crypto-js": "^3.1.9-1",
    "expo": "^20.1.3",
    "pusher-js": "^4.1.0",
    "qs": "^6.5.0",
    "react": "16.0.0-alpha.12",
    "react-native": "https://github.com/expo/react-native/archive/sdk-20.0.0.tar.gz",
    "react-native-app-link": "^0.4.0",
    "react-native-dotenv": "^0.1.0",
    "react-native-modal": "^3.1.0",
    "react-navigation": "^1.0.0-beta.12"
  }
}
```
**app.json:**
```
{
	"expo": {
		"sdkVersion": "20.0.0",
		"name": "Pocat",
		"slug": "pocat",
		"version": "1.0",
		"icon": "https://s3.amazonaws.com/pokerbuddy/images/app_icon.png",
		"primaryColor": "#bc0000",
		"orientation": "portrait",
		"ios": {
			"bundleIdentifier": "com.hecodesthings.pocat",
			"buildNumber": "2"
		},
		"android": {
			"package": "com.hecodesthings.pocat",
			"versionCode": 2
		},
		"scheme": "pocat",
		"facebookScheme": "fb309462139518578"
	}
}
```

**Comment by jimmylee** [comment]
Pinging @jesseruder to have a look at this, I'm personally not sure what the issue is.

**Comment by rotembcohen** [comment]
Thank you. I'll also add the following resources:
[github code of the app](https://github.com/rotembcohen/pokerBuddyApp)
[link to the expo listing](https://expo.io/@rotembcohen/pocat)
[the app on the google play store](https://play.google.com/store/apps/details?id=com.hecodesthings.pocat)

**Comment by EsdrasEtrenne** [comment]
Im having this issue a ton lately too and I have no clue where it's coming from. There is no real details to the error and not a ton of documentation on it

**Comment by EsdrasEtrenne** [comment]
Im using an HTC M9 android device, react native .48.4, expo sdk 21

```
IllegalViewOperationException: Trying to add unknown view tag: 80
detail: View tag: 72
children(2):[
73,75,
],
viewsToAdd(1):[
[2,80],
]

```
This crash almost always happens at startup for my app. If this is not the error then sometimes there is an error that reads:
` Error while updating property 'xxx' in shadow node of type RCTView. Sometimes it is `RCTImageView

**Comment by rotembcohen** [comment]
@EsdrasEtrenne
It's not just you, I've also been experiencing that one:

abi20_0_0.com.facebook.react.uimanager.IllegalViewOperationException: Trying to add unknown view tag: 13
detail: View tag:9
children(1): [
12,
],
viewsToAdd(1): [
[1,13],
],
at abi20_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi20_0_0.com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:900)
  at abi20_0_0.com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:31)
  at abi20_0_0.com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:136)
  at abi20_0_0.com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:107)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:957)
  at android.view.Choreographer.doCallbacks (Choreographer.java:734)
  at android.view.Choreographer.doFrame (Choreographer.java:667)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:945)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6776)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1520)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1410)

**Comment by EsdrasEtrenne** [comment]
I thought this might be related

https://stackoverflow.com/questions/45002568/com-facebook-react-uimanager-illegalviewoperationexception-trying-to-add-unknow

But my app is upgraded past react native .48 now.

I've also seen a lot of issues surrounding animation. I did have one of my first screens using an animation so I removed that animation from the app and the issue is still firing.

The only other place I have any reference to animation in the app is the root component where I set `animationEnabled: false` a few times to my navigators. So I don't think animation is the issue at least in my case.

The only other thing I can think of is maybe react native doesn't like that I have my root component wrapped in a view tag?

I do it so that I can give a little padding to every screen if the screen is Android
```


   return (
     <Provider store = {store}>
       <View style={styles.container}>
         <MenuContext>
           <MainNavigator
             onNavigationStateChange={(prevState, currentState, action) => {
                 getCurrentRouteName(currentState);
                 const defaultGetStateForAction = MainNavigator.router.getStateForAction;
                 const getRouteName = (state) => {
                   const route = state.routes[state.index];
                   return typeof route.index === 'undefined' ? route.routeName : getRouteName(route);
                 }
                 MainNavigator.router.getStateForAction = (action, state) => {
                   if(action.routeName == getRouteName(state)){
                     return state;
                   }
                   return defaultGetStateForAction(action, state);
                 };
               }
             }
           />
         </MenuContext>
       </View>
     </Provider>
   );

const styles = StyleSheet.create({
 container: {
   flex: 1,
   backgroundColor: '#fff',
   justifyContent: 'center',
   paddingTop: Platform.OS === "android"? (Constants.statusBarHeight): 0
 },
});

Expo.registerRootComponent(App);
```

I can experiment later with removing that and just adding padding to each individual screen.

I only think this might be the issue because in a few cases like the stackoverflow solution above the rootView is mentioned. Apparently the` VIEW_ID `of the root view should be empty, but it isn't really clear.


Also found this definition of the error for reference:
https://www.cosc.brocku.ca/sites/all/files/documentation/Brock_packages/BasicIO/IllegalOperationException.html

**Comment by rotembcohen** [comment]
I can confirm that it's unrelated to animation since I don't use animation at all. Also, it only happens in Android for me and the only difference between Android and iOS in my code are a few styling differences, so I assume this is really an expo Android client thing, and not a react native thing.

**Comment by rotembcohen** [comment]
Sorry, closed by mistake

**Comment by EsdrasEtrenne** [comment]
Maybe, but since I'm not seeing this problem reported very heavily I wonder if it could be something I'm running uniquely that is somehow uncommon or incorrect. Android and iOS work differently so some things will be ok on iOS and wont fly on Android

**Comment by rotembcohen** [comment]
Well, currently I'm getting this error only on one device. All my other Android users are not experiencing this error as of now

**Comment by EsdrasEtrenne** [comment]
Oh that's really interesting because I'm the only one getting the issue too. So far anyway, I was concerned more would eventually have it when we leave beta

**Comment by rotembcohen** [comment]
I only have a very few users (currently 9) on android so I don't know how much this can say. Maybe it's something that is caused bc of the app installed together with the expo client on the same device? Weird that only the developers get this. But if it's like that, then I'm less worried actually.

**Comment by dailycommit** [comment]
I am also struggling with this problem.
The application never opens again in the background.
Does this have a solution?
I am getting errors like below, or the application is completely gone and I need to open it back.

![screenshot_20171002-163400](https://user-images.githubusercontent.com/31360927/31197227-20a6c6e0-a959-11e7-992a-3f78002bfa92.png)
![screenshot_20171003-033051](https://user-images.githubusercontent.com/31360927/31197228-20ac77b6-a959-11e7-8862-8775fe91ee3c.png)

**Comment by jbakerdev** [comment]
@dailycommit In my experience, I've seen those when animating a style property to a weird value i.e. trying to transition top from 0 to '100%' (integer to string) or similar.

**Comment by EsdrasEtrenne** [comment]
@dailycommit 

I was actually getting those exact same issues but I havent seen them as of recently. I also havent had my app crash with this issue for a while although it's too soon to say it is not happening anymore. 

One thing that I did with this is to make sure that all of my styles were whole integer numbers. Also, when I did need a thin line, I used the inbuilt hairLineWidth with react native:
(hairlineWidth can only be used by also using StyleSheet

https://facebook.github.io/react-native/docs/stylesheet.html
```
const dark_styles = StyleSheet.create({
  lineStyle:{
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor:'#31496A',
    marginRight: Math.ceil(SCREEN_WIDTH*.01),
    marginLeft: Math.ceil(SCREEN_WIDTH*.01),
  },
  contactStyle:{
    flexDirection:'row',
    marginTop: Math.ceil(SCREEN_HEIGHT*.02),
    marginBottom: Math.ceil(SCREEN_HEIGHT*.02),
    marginLeft: Math.ceil(SCREEN_WIDTH*.02),
  },
  textStyle:{
    marginLeft: Math.ceil(SCREEN_HEIGHT*.01),
    fontSize: 14,
    color: "#F6F3E7",
    justifyContent: "center",
    fontFamily: "kohinoor",
  },
  usernameStyle:{
    marginLeft: Math.ceil(SCREEN_HEIGHT*.01),
    color: "gray",
    fontSize: 12,
    justifyContent: "center",
    fontFamily: "kohinoor",
  },
})
```

I dont know how big of a deal it was, but I read somewhere that using decimal or floats for values can cause issues

**Comment by dailycommit** [comment]
Thank you @jbakerdev @EsdrasEtrenne, 

I'm working on a simple application and I have hardly ever created my own style. I am using [shoutem-ui](https://github.com/shoutem/ui) toolkit for the interface. If it is not related to expo, do you think this shoutem-ui issue?

Is there any way to find out exactly what the problem is?  which style or component is giving an error. Where can I see the complete error report?

**Comment by dikaiosune** [comment]
@rotembcohen I just tried your repro steps on a regular s8, also running android 7, and wasn't able to trigger this crash. Have you published new code for your play store binary recently?

It seems pretty common for community libraries and app code to trigger `IllegalViewOperationException` with React Native, and AFAIK once your app is running we don't touch the view hierarchy at all. Since I can't repro with your current application, I'm not sure how to proceed.

@dailycommit given the existing bug reports about these exceptions, I wouldn't be surprised if shoutem put your native-side view hierarchy into an invalid state at some point. That said without a specific reproduction we can test, it's difficult to say for sure where the problem is.

Going to give this a close since I'm not able to reproduce, but if it's still happening for people please provide some info we can use to track this down and I'll happily reopen. Thanks!

**Comment by EsdrasEtrenne** [comment]
So we just went into production with our app and this is still an issue. It is not just me but other users as well getting the error who use the app.


First Error:
```
abi21_0_0.com.facebook.react.uimanager.IllegalViewOperationException: 
  at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.resolveViewManager (NativeViewHierarchyManager.java:109)
  at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren (NativeViewHierarchyManager.java:265)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded (UIViewOperationQueue.java:813)
  at abi21_0_0.com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:21)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:173)
  at android.app.ActivityThread.main (ActivityThread.java:6523)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:938)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:828)
```

Second Error:
```
abi21_0_0.com.facebook.react.uimanager.IllegalViewOperationException: Trying to add unknown view tag: 57
detail: View tag:56
children(0): [
],
viewsToAdd(1): [
[0,57],
],
at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:900)
  at abi21_0_0.com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:31)
  at abi21_0_0.com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:136)
  at abi21_0_0.com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:107)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:759)
  at android.view.Choreographer.doCallbacks (Choreographer.java:574)
  at android.view.Choreographer.doFrame (Choreographer.java:543)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:747)
  at android.os.Handler.handleCallback (Handler.java:733)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:136)
  at android.app.ActivityThread.main (ActivityThread.java:5584)
  at java.lang.reflect.Method.invokeNative (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:515)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1268)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1084)
  at dalvik.system.NativeStart.main (Native Method)
```


If the error were more resourceful I could find out where the issue is with the code if there is any but I  do not see anything pointing to the code.

**Comment by tiaaaa123** [comment]
I'm getting the same crash in production with my app. Lots of them

In the last 7 days
```
126 crashes

abi21_0_0.com.facebook.react.uimanager.IllegalViewOperationException: 
  at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.resolveViewManager (NativeViewHierarchyManager.java:109)
  at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren (NativeViewHierarchyManager.java:265)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded (UIViewOperationQueue.java:813)
  at abi21_0_0.com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:21)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6688)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1468)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1358)
```

```
106 crashes

abi21_0_0.com.facebook.react.uimanager.IllegalViewOperationException: Trying to add unknown view tag: 32
detail: View tag:12
children(1): [
15,
],
viewsToAdd(1): [
[1,32],
],
at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:900)
  at abi21_0_0.com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:31)
  at abi21_0_0.com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:136)
  at abi21_0_0.com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:107)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:869)
  at android.view.Choreographer.doCallbacks (Choreographer.java:683)
  at android.view.Choreographer.doFrame (Choreographer.java:616)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:857)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6247)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:872)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:762)
```

and 

```
94 crashes

abi21_0_0.com.facebook.react.uimanager.IllegalViewOperationException: Trying to add unknown view tag: 26
detail: View tag:15
children(2): [
20,24,
],
viewsToAdd(1): [
[2,26],
],
at abi21_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:177)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:776)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:855)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$1600 (UIViewOperationQueue.java:46)
  at abi21_0_0.com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded (UIViewOperationQueue.java:813)
  at abi21_0_0.com.facebook.react.bridge.GuardedRunnable.run (GuardedRunnable.java:21)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6682)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1520)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1410)
```

App crashing upon opening or crashing when the app is put in the background for a while...

Dependencies: 
```
"@bam.tech/react-native-root-siblings": "^1.2.1",
    "@expo/ex-navigation": "~3.0.0",
    "deepmerge": "^1.3.2",
    "expo": "^21.0.0",
    "immutable": "^3.8.1",
    "lodash.debounce": "^4.0.8",
    "moment": "^2.18.1",
    "moment-duration-format": "^1.3.0",
    "moment-timezone": "^0.5.13",
    "react": "16.0.0-alpha.12",
    "react-native": "https://github.com/expo/react-native/archive/sdk-21.0.2.tar.gz",
    "react-native-animatable": "^1.2.4",
    "react-native-autocomplete-input": "3.1.0",
    "react-native-autolink": "^1.0.0",
    "react-native-flip-card": "3.4.1",
    "react-native-i18n": "^1.0.0",
    "react-native-swipeable": "^0.6.0",
    "react-native-swiper": "^1.5.10",
    "react-native-vector-icons": "^4.0.1",
    "react-redux": "^5.0.4",
    "redux": "^3.6.0",
    "redux-mock-store": "^1.2.3",
    "redux-thunk": "^2.2.0",
    "reselect": "^3.0.0",
    "sentry-expo": "^1.6.0"
```

**Comment by EsdrasEtrenne** [comment]
Here are my dependencies as well. I really don't think this should be closed just because it is difficult to recreate. If you use the app for about a week and open it enough times in the day the error will pop up and only for android. My iOS users haven't gotten this.

Also, always at start up or when app is in the background for a long time

```
    "axios": "^0.16.2",
    "expo": "^21.0.0",
    "lodash": "^4.17.4",
    "moment": "^2.18.1",
    "react": "16.0.0-alpha.12",
    "react-native": "https://github.com/expo/react-native/archive/sdk-21.0.2.tar.gz",
    "react-native-actioncable": "0.0.3",
    "react-native-animatable": "^1.2.3",
    "react-native-datepicker": "^1.5.1",
    "react-native-elements": "git://github.com/react-native-training/react-native-elements.git#next",
    "react-native-geocoding": "^0.2.0",
    "react-native-google-shortener": "^1.1.0",
    "react-native-hyperlink": "0.0.10",
    "react-native-image-to-base64": "^0.1.0",
    "react-native-keyboard-aware-scroll-view": "^0.3.0",
    "react-native-modal-dropdown": "^0.5.0",
    "react-native-popup-menu": "^0.7.5",
    "react-navigation": "^1.0.0-beta.11",
    "react-redux": "^5.0.5",
    "redux": "^3.6.0",
    "redux-persist": "^4.8.0",
    "redux-thunk": "^2.2.0",
    "sentry-expo": "~1.6.0"
```

**Comment by rotembcohen** [comment]
@dikaiosune Hey, Adam. Thanks for looking into this, but unfortunately, it still happens (happened to me this morning).

<img width="995" alt="screen shot 2017-10-16 at 2 56 50 pm" src="https://user-images.githubusercontent.com/5367839/31629669-52b79e20-b282-11e7-8aec-5b0528dbbe46.png">

**Comment by dikaiosune** [comment]
Hi everyone! After poking around, it looks like at least one cause of this bug in upstream React Native will be fixed when we release support for RN 0.49 (in SDK 22). Stay tuned for that announcement and please post back if you're still seeing after upgrading.

**Comment by EsdrasEtrenne** [comment]
I just upgraded the app Im waiting to see if I still get it. Thank you

**Comment by EsdrasEtrenne** [comment]
@dikaiosune I did upgrade the code to sdk 22 and still getting the error.
```

abi22_0_0.com.facebook.react.uimanager.IllegalViewOperationException: Trying to add unknown view tag: 34
detail: View tag:35
children(0): [
],
viewsToAdd(1): [
[0,34],
],
at abi22_0_0.com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java
  at abi22_0_0.com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute (UIViewOperationQueue.java:180)
  at abi22_0_0.com.facebook.react.uimanager.UIViewOperationQueue$1.run (UIViewOperationQueue.java:815)
  at abi22_0_0.com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches (UIViewOperationQueue.java:922)
  at abi22_0_0.com.facebook.react.uimanager.UIViewOperationQueue.access$2100 (UIViewOperationQueue.java:47)
  at abi22_0_0.com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded (UIViewOperationQueue.java:982)
  at abi22_0_0.com.facebook.react.uimanager.GuardedFrameCallback.doFrame (GuardedFrameCallback.java:31)
  at abi22_0_0.com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame (ReactChoreographer.java:136)
  at abi22_0_0.com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame (ChoreographerCompat.java:107)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:957)
  at android.view.Choreographer.doCallbacks (Choreographer.java:734)
  at android.view.Choreographer.doFrame (Choreographer.java:667)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:945)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6776)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1386)
```

I am getting a few errors and I really have no cue how to go about solving them besides googling the error. There are no references to a piece of the code that has gone wrong, just very arbitrary error details that are either mentioning java or facebook.react


Is there something I can do to better debug this on my own are these issues mostly a react issue?

**Comment by kareem-adel** [comment]
@EsdrasEtrenne I was facing a similar issue, after looking up the problem it seems like the views are being removed from the native view hierarchy as an optimization, therefore to make sure that the view is present in the view hierarchy set the prop collapsable to false - in my case in the root view -.
`collapsable={false}`
Reference http://facebook.github.io/react-native/releases/0.49/docs/viewproptypes.html#collapsable

**Comment by EsdrasEtrenne** [comment]
Dear God, so much very thank you. 

I will throw that in and see what happens. Makes sense since it is not a iOS issue.

**Comment by mathieumg** [comment]
Testing the same thing here and will report back, thanks @kareem-adel .

**Comment by EsdrasEtrenne** [comment]
Update: Still getting the error after just applying `collapsable = {false}` to the root view. Going to apply it to the main container view of each Screen that is rendered and see.

**Comment by tiaaaa123** [comment]
@EsdrasEtrenne Collapsable seems to be a synonym of Collapsible. So it's not a typo.
But I'm also interested to this fix. Will try in my project too

**Comment by EsdrasEtrenne** [comment]
Update: seem to be getting this error much less but still occurring. I will try putting the collapsable false on all view tags. This seems overkill but we will see

**Comment by EsdrasEtrenne** [comment]
So apparently a similar bug was fixed in #779 after updating to SDK 23. I'm waiting on horizontal scrolling to be fixed before upgrading to see if this will fix our issue too. SDK 23 has a few things I will need to fix before upgrading so I cant check on this immediately. If anyone else can that'd be great :)

**Comment by mikesholiu** [comment]
I am getting this issue as well and on Expo SDK 24. Have not tried the`collapsable` fix yet. @EsdrasEtrenne are you still having this issue with your latest build?

**Comment by otaviogaiao** [comment]
I was having a similar problem. It always happened in one screen that had a modal... I only tested on one Android device, so I don't know if other devices would suffer from it. iOS was fine.

![screenshot_2018-02-01-21-16-10-020_com miui bugreport](https://user-images.githubusercontent.com/3965485/35709763-2c73a088-079b-11e8-8e0e-ce971faac6c1.png)

It turns out it was being caused by one of my styles using a height of '15.03%'. Once I changed it to '15%', the problem went away and it didn't happen again. Honestly, I have no idea why this happened... I use % a lot in this app and it never happened before. I tried changing to '10.03%' and other values keeping the '.03%' and the problem persisted. Only once I removed the '.03%' it went away. Really really weird.
